10.1/s2i/bin/assemble

```diff
@@ -275,6 +275,22 @@ if [ -f "$LOCAL_SOURCE_DIR/pom.xml" ]; then
   if [ -n "$MAVEN_ARGS_APPEND" ]; then
     export MAVEN_ARGS="$MAVEN_ARGS $MAVEN_ARGS_APPEND"
   fi
+
+  CONTAINER_HEAP_PERCENT=${CONTAINER_HEAP_PERCENT:-0.50}
+  CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+  CONTAINER_MEMORY_IN_MB=$((${CONTAINER_MEMORY_IN_BYTES}/1024**2))
+  CONTAINER_HEAP_MAX=$(echo "${CONTAINER_MEMORY_IN_MB} ${CONTAINER_HEAP_PERCENT}" | awk '{ printf "%d", $1 * $2 }')
+  JAVA_HEAP_MAX=-Xmx${CONTAINER_HEAP_MAX}
```

Comment:
it should be gated on the 2**40 upper bound, i was thinking that was related to the 32/64bit heap logic and left it out, but that was incorrect.  i'll put it back in.

As for -Xms, i did not want to set the minimum heap size because I was worried about increasing the minimal footprint for a maven compilation and possibly causing people's builds to start failing if there was enough memory to run the build, but not enough if the full heap was immediately claimed.  (which would imply the max heap value is also too large, but if the build doesn't allocate the full heap, you'd get away with it).